packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-header/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { css, spacing, Body, Icon } from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import PipelineStages from './pipeline-stages';
+import PipelineActions from './pipeline-actions';
+import {
+  getSavedPipelines,
+  savedPipelinesListToggle,
+} from '../../../modules/saved-pipeline';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'space-between',
+});
+
+const pipelineTextAndStagesStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+const pipelineStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  marginRight: spacing[5],
+});
+
+const pipelineTextStyles = css({
+  marginRight: spacing[1],
+  marginLeft: spacing[1],
+});
+
+const openSavedPipelinesStyles = css({
+  display: 'flex',
+  marginRight: spacing[1],
+  marginLeft: spacing[1],
+  cursor: 'pointer',
+});
+
+const PipelineHeader: React.FunctionComponent<PipelineHeaderProps> = ({
+  onShowSavedPipelines,
+}) => {
+  return (
+    <div className={containerStyles}>
+      <div className={pipelineTextAndStagesStyles}>
+        <div className={pipelineStyles}>
+          <Body weight="medium" className={pipelineTextStyles}>
+            Pipeline
+          </Body>
+          <div
+            role="button"
+            onClick={() => onShowSavedPipelines()}
+            className={openSavedPipelinesStyles}
+            aria-label="Open saved pipelines"
+            tabIndex={0}
+          >
+            <Icon glyph="Folder" />
+            <Icon glyph="CaretDown" />
+          </div>
+        </div>
+        <PipelineStages />
```

Comment:
This group of elements has its spacing definitions a bit all over the place, you have this pipeline button have most of the margins responsible for spacing here, but then the pipeline has it's own small margin (that will not stack because this is a flex container)

<img width="193" alt="image" src="https://user-images.githubusercontent.com/5036933/158769408-47ff6da2-5ea9-4944-bc99-3fe18581e9bd.png">
<img width="263" alt="image" src="https://user-images.githubusercontent.com/5036933/158769426-544e22a5-8d4d-4c9b-9a07-4cebb6f25344.png">

As a rule of thumb I really suggest you to aim for defining spacing in a way that will not mess with the layout if elements are moved around, used in another context, or removed completely (it's really conceptually similar to this other comment I left). In this case I would suggest that you remove the margin from the pipeline component and wrap it in something and have a spacing for it defined here, that way not only your pipeline component can be moved around easier, you also have one place that is responsible for defining layout of the toolbar, not three like it is right now